src/main/java/com/spotify/reaper/service/SegmentGenerator.java

```diff
@@ -156,7 +157,7 @@ public static boolean greaterThanOrEqual(BigInteger a, BigInteger b) {
       BigInteger size = segment.span(RANGE_SIZE);
       total = total.add(size);
     }
-    if (!total.equals(RANGE_SIZE)) {
+    if (!total.equals(RANGE_SIZE) && !incrementalRepair) {
```

Comment:
Is this necessarily the case? And feels weird that this is the only place we use incrementalRepair argument.